keysign/util.py

```diff
@@ -85,6 +149,58 @@ def email_file(to, from_=None, subject=None,
     return retval
 
 
+def _using_flatpak():
+    """Check if we are inside flatpak"""
+    return os.path.exists("/.flatpak-info")
+
+
+def _fix_path_flatpak(files):
+    """In Flatpak the only special path visible also from outside is /var/tmp/
+    To be able to use the files from the host we change the path to the absolute one.
+    This fix in the future may not be necessary because the portals should be able
+    to automatically handle it."""
+    tmp_flat = "/tmp"
+    var_flat = "/var"
+    var_tmp_flat = "/var/tmp"
+    part_1 = os.path.expanduser("~/.var/app/")
+    app_id = "org.gnome.Keysign"
+    part_2 = "cache/tmp/"
+    flatpak_path = os.path.join(part_1, app_id, part_2)
+    fixed_files = []
+    if files:
+        for file in files:
+            if file.startswith(tmp_flat):
+                shutil.move(file, var_flat + file)
```

Comment:
woahr. is this not a crazy hack that depends on the full file path starting with something that is also writeable in `/var`?  If, for some odd reason, the file is named "/export/tmp/foo.asc" then this construction will attempt to create `/var/export/foo.asc`, no?
This is not good.

Also, use `os.path.join`.

How about taking the basename of the file and then move (or maybe even copy only) it to `$XDG_CACHE_HOME/tmp` or something..? Where `$XDG_CACHE_HOME` should probably fall back to `~/.var/app/$ID/cache`.